libgnucash/engine/test/gtest-gnc-datetime.cpp

```diff
@@ -182,7 +221,7 @@ TEST(gnc_date_constructors, test_str_format_constructor)
             got_month = test_ymd.month;
             got_day = test_ymd.day;
         }
-        catch (const std::invalid_argument& e)
+        catch (...)
```

Comment:
At least use `catch(std::logic_error)`. That's the base class of both std::invalid_argument and std::out_of_range, the two exceptions that boost::date_time's parser throws. Better would be to have a catch for each so that a std::length_error or std::domain_error will still crash. You could go so far as to catch each of the boost::date_time std::out_of_range derived classes, but that's getting a bit verbose.